CMakeLists.txt

```diff
@@ -380,33 +379,46 @@ if(DEBUG OR "${CMAKE_BUILD_TYPE}" STREQUAL "Debug")
     add_test(NAME moose.bin-raw-run COMMAND moose.bin -u -q)
 endif()
 
-## PyMOOSE tests.
-set(PYMOOSE_TEST_DIRECTORY ${CMAKE_SOURCE_DIR}/tests/python)
-
-# Collect all python script in tests folder and run them.
+# PyMOOSE tests. These tests complete in reasonable time and should run by
+# default.
+set(PYMOOSE_TEST_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/tests/py_moose)
 file(GLOB PY_TEST_SCRIPTS "${PYMOOSE_TEST_DIRECTORY}/test_*.py" )
-
-
-# Run python tests.
-foreach( _test_script ${PY_TEST_SCRIPTS} )
-    get_filename_component( _test_name ${_test_script} NAME_WE)
-    add_test( NAME ${_test_name}
+foreach(_test_script ${PY_TEST_SCRIPTS} )
+    get_filename_component(_name_we ${_test_script} NAME_WE)
+    set(_test_name "pymoose_${_name_we}")
+    add_test(NAME ${_test_name}
         COMMAND ${PYTHON_EXECUTABLE} ${_test_script}
         WORKING_DIRECTORY ${PYMOOSE_TEST_DIRECTORY}
         )
-     set_tests_properties( ${_test_name}
+     set_tests_properties(${_test_name}
          PROPERTIES ENVIRONMENT "PYTHONPATH=${CMAKE_CURRENT_BINARY_DIR}/python"
         )
 endforeach( )
 
-# Alpha tests. These should not run by default.
-set(MOOSE_ALPHA_TEST_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/tests/alpha)
-file(GLOB PY_ALPHA_TEST_SCRIPTS "${PYMOOSE_ALPHA_TEST_DIRECTORY}/*.py" )
-foreach( _test_script ${PY_ALPHA_TEST_SCRIPTS} )
-    get_filename_component( _test_name ${_test_script} NAME_WE)
+# rdesigneur tests. These tests require matplotlib.
+set(RDES_TEST_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/tests/py_rdesigneur) 
+file(GLOB RDES_TEST_SCRIPTS "${RDES_TEST_DIRECTORY}/test_*.py" )
+foreach(_test_script ${RDES_TEST_SCRIPTS})
+    get_filename_component(_name_we ${_test_script} NAME_WE)
+    set(_test_name "rdes_${_name_we}")
+    add_test(NAME ${_test_name}
+        COMMAND ${PYTHON_EXECUTABLE} ${_test_script}
+        WORKING_DIRECTORY ${RDES_TEST_DIRECTORY}
+        )
+    set_tests_properties(${_test_name} PROPERTIES 
+        ENVIRONMENT "PYTHONPATH=${CMAKE_CURRENT_BINARY_DIR}/python;MOOSE_NUM_THREADS=4"
+        )
+endforeach()
+
+# FIXME TESTS. These should not run by default. We need to fix them.
```

Comment:
These tests in `fixme` directory requires fixing sooner or later. They don't run by default. To run them use `-C fixme` with `ctest`.